ocode_python/core/semantic_context.py

```
@@ -0,0 +1,762 @@
+
+        try:
+            # Get query embedding
+            query_embedding = self.embeddings_model.encode(query)
+
+            # Process files in batches to avoid memory issues
+            batch_size = 10
+            for i in range(0, len(semantic_files), batch_size):
+                batch = semantic_files[i : i + batch_size]
+                await self._process_embedding_batch(query_embedding, batch)
+
+        except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)